pkg/detectors/alchemy/alchemy.go

```diff
@@ -31,57 +32,69 @@ func (s Scanner) Keywords() []string {
 	return []string{"alchemy"}
 }
 
-// FromData will find and optionally verify Alchemy secrets in a given set of bytes.
+// FromData will find and optionally verifyMatch Alchemy secrets in a given set of bytes.
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
 	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
-
-	for _, match := range matches {
-		if len(match) != 2 {
```

Comment:
This doesn't actually do what it's meant to, and will cause silent failures if the number of match groups is changed.

https://github.com/trufflesecurity/trufflehog/pull/2264#discussion_r1443382434